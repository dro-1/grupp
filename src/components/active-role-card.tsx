import roleIcon from "@/assets/icons/role.svg";
import check from "@/assets/icons/check.svg";
import { twMerge } from "tailwind-merge";
import { ActiveRole } from "@/types/types";

type RoleCardProps = {
  activeRole: ActiveRole;
  selected: boolean;
  onClick: () => void;
};

export const ActiveRoleCard: React.FC<RoleCardProps> = ({
  activeRole,
  selected,
  onClick,
}) => {
  return (
    <div
      onClick={onClick}
      className={twMerge(
        "flex p-4 bg-white border border-gray-200 rounded-lg",
        selected && "border-primary-300 bg-primary-50"
      )}
    >
      <div className="bg-white w-[45px] h-[31px] border border-[#f2f4f7] flex justify-center items-center rounded-[6px] mr-3">
        <img src={roleIcon} alt="" />
      </div>
      <div className="grow tracking-0 flex flex-col">
        <em
          className={twMerge(
            "not-italic font-medium text-sm text-gray-700",
            selected && "text-primary-800"
          )}
        >
          {activeRole.roleName}
        </em>
        <em
          className={twMerge(
            "not-italic text-[#70798C] text-sm",
            selected && "text-primary-600"
          )}
        >
          Last active {activeRole.lastActiveDate}
        </em>
        <div className="mt-2">
          <button
            className={twMerge(
              "text-gray-500 font-medium text-sm mr-3",
              selected && "text-primary-500"
            )}
          >
            Set as default
          </button>
          <button className="text-primary-700 font-medium text-sm">Edit</button>
        </div>
      </div>
      <div
        className={twMerge(
          "w-4 h-4 bg-white border border-gray-300 rounded-full flex justify-center items-center transition",
          selected && "bg-primary-600 border-primary-600"
        )}
      >
        <img src={check} alt="" />
      </div>
    </div>
  );
};
