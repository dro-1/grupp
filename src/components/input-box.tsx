import { twMerge } from "tailwind-merge";

type InputBoxProps = {
  iconSrc: string;
  placeholder: string;
  iconClassname?: string;
  className?: string;
};

export const InputBox: React.FC<InputBoxProps> = ({
  iconSrc,
  placeholder,
  iconClassname,
  className,
}) => {
  return (
    <div
      className={twMerge(
        "mt-3 relative rounded-md transition-shadow duration-300 focus-within:shadow-[0_4px_0_0_#f4ebff,0_-4px_0_0_#f4ebff,4px_0_0_0_#f4ebff,-4px_0_0_0_#f4ebff]",
        className
      )}
      data-testid="input-box"
    >
      <img
        className={twMerge(
          "absolute top-[15.33px] left-[15.67px] z-20",
          iconClassname
        )}
        src={iconSrc}
        alt=""
      />
      <input
        className="peer px-[14px] py-[10px] bg-white border border-gray-300 rounded-lg w-full text-gray-900 pl-[42px] outline-none transition focus:border-primary-500 relative z-10"
        placeholder={placeholder}
      />
    </div>
  );
};
