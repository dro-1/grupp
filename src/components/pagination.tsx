import { twMerge } from "tailwind-merge";

type PaginationProps = {
  page: number;
  setPage: (page: number) => void;
  startPage: number;
  endPage: number;
  middlePages: number[];
  totalPages: number;
};

export const Pagination: React.FC<PaginationProps> = ({
  page,
  setPage,
  startPage,
  endPage,
  middlePages,
  totalPages,
}) => (
  <div className="flex justify-center mt-6 gap-2">
    <button
      key={1}
      className={twMerge(
        "px-3 py-1 rounded border border-gray-300 bg-white text-gray-700 font-medium text-sm transition lg:hover:bg-primary-50 lg:hover:border-primary-600",
        page === 1 && "bg-primary-50 border-primary-600 text-primary-700"
      )}
      onClick={() => setPage(1)}
    >
      1
    </button>
    {startPage > 2 && <span className="px-2">...</span>}
    {middlePages.map((num) => (
      <button
        key={num}
        className={twMerge(
          "px-3 py-1 rounded border border-gray-300 bg-white text-gray-700 font-medium text-sm transition lg:hover:bg-primary-50 lg:hover:border-primary-600",
          page === num && "bg-primary-50 border-primary-600 text-primary-700"
        )}
        onClick={() => setPage(num)}
      >
        {num}
      </button>
    ))}
    {endPage < totalPages - 1 && <span className="px-2">...</span>}
    <button
      key={totalPages}
      className={twMerge(
        "px-3 py-1 rounded border border-gray-300 bg-white text-gray-700 font-medium text-sm transition lg:hover:bg-primary-50 lg:hover:border-primary-600",
        page === totalPages &&
          "bg-primary-50 border-primary-600 text-primary-700"
      )}
      onClick={() => setPage(totalPages)}
    >
      {totalPages}
    </button>
  </div>
);
