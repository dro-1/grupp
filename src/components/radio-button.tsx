import { ReactNode, useRef } from "react";

type RadioButtonProps = {
  content: ReactNode;
  name: string;
  id: string;
};

export const RadioButton: React.FC<RadioButtonProps> = ({
  content,
  name,
  id,
}) => {
  const inputRef = useRef<HTMLInputElement>(null);
  const handleClick = () => {
    inputRef.current!.checked = true;
  };
  return (
    <label onClick={handleClick} className="flex items-start" htmlFor={id}>
      <input
        ref={inputRef}
        type="radio"
        className="peer hidden"
        name={name}
        id={id}
      />

      <div className="w-4 h-4 border cursor-pointer border-gray-300 rounded-full transition bg-white mr-2 relative after:absolute after:w-[6px] after:h-[6px] after:bg-primary-600 after:rounded-full after:top-1/2 after:left-1/2 after:-translate-1/2 after:opacity-0 after:transition lg:hover:bg-primary-50 lg:hover:border-primary-600 peer-checked:bg-primary-50 peer-checked:border-primary-600 peer-checked:after:opacity-100" />
      {content}
    </label>
  );
};
