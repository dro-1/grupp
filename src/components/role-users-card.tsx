export const RoleUsersCard: React.FC<{ images: string[] }> = ({ images }) => {
  return (
    <div className="flex w-[180px] h-[40px] relative">
      {images.slice(0, 5).map((image, idx) => (
        <img
          key={idx}
          className="w-10 h-10 absolute top-0"
          src={image}
          alt=""
          style={{
            left: 28 * idx,
          }}
        />
      ))}
      {images.length > 5 && (
        <div
          className="relative w-10 h-10 bg-gray-50 border-2 border-white rounded-full flex justify-center items-center font-medium text-gray-600"
          style={{ left: 28 * 5 }}
        >
          +{images.length - 5}
        </div>
      )}
    </div>
  );
};
