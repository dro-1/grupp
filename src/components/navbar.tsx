import { useState } from "react";
import { twMerge } from "tailwind-merge";
import { InputBox } from "./input-box";

import logo from "@/assets/icons/logo.svg";
import hamburger from "@/assets/icons/hamburger.svg";
import home from "@/assets/icons/home.svg";
import dashboard from "@/assets/icons/dashboard.svg";
import projects from "@/assets/icons/projects.svg";
import tasks from "@/assets/icons/tasks.svg";
import reporting from "@/assets/icons/reporting.svg";
import users from "@/assets/icons/users.svg";
import support from "@/assets/icons/support.svg";
import settings from "@/assets/icons/settings.svg";
import search from "@/assets/icons/search.svg";
import logout from "@/assets/icons/logout.svg";

import user1 from "@/assets/images/user-1.png";
import navImage from "@/assets/images/nav-image.png";

const topLinks = [
  {
    title: "Home",
    icon: home,
  },
  {
    title: "Dashboard",
    icon: dashboard,
    count: 10,
  },
  {
    title: "Projects",
    icon: projects,
  },
  {
    title: "Tasks",
    icon: tasks,
  },
  {
    title: "Reporting",
    icon: reporting,
  },
  {
    title: "Users",
    icon: users,
  },
];

const bottomLinks = [
  {
    title: "Support",
    icon: support,
  },
  {
    title: "Settings",
    icon: settings,
  },
];

export const Navbar = () => {
  const [sidebarOpen, setSidebarOpen] = useState(false);
  return (
    <>
      <div className="w-full fixed top-0 z-30 lg:hidden">
        {" "}
        <nav className="p-4 flex justify-between items-center bg-white ">
          <div className="flex items-center">
            <img className="block mr-[10px]" src={logo} alt="" />
          </div>
          <button onClick={() => setSidebarOpen(true)}>
            <img src={hamburger} alt="" />
          </button>
        </nav>
        <div className="w-full h-[1px] bg-gray-200" />
      </div>

      {/* Sidebar overlay with transition, opacity, and pointer-events controlled by sidebarOpen */}
      <div
        className={twMerge(
          "fixed inset-0 z-40 bg-black/30 transition-opacity duration-300 lg:hidden",
          sidebarOpen
            ? "opacity-100 pointer-events-auto"
            : "opacity-0 pointer-events-none"
        )}
        onClick={() => setSidebarOpen(false)}
        tabIndex={-1}
        aria-hidden={!sidebarOpen}
        role="button"
      />
      {/* Sidebar menu */}
      <aside
        className={twMerge(
          "fixed top-0 left-0 h-full w-[80vw] max-w-[400px] bg-white shadow-lg z-50 transform transition-transform duration-300 flex flex-col lg:translate-x-0 lg:static lg:w-[280px] lg:h-screen lg:overflow-y-auto lg:shrink-0",
          sidebarOpen ? "translate-x-0" : "-translate-x-full"
        )}
        tabIndex={-1}
        aria-hidden={!sidebarOpen}
      >
        <div className="p-4 flex justify-between items-center">
          <img className="block mr-[10px]" src={logo} alt="" />
        </div>
        <div className="px-4">
          <InputBox
            iconSrc={search}
            placeholder="Olivia Rhye"
            iconClassname="top-3 left-[14px]"
            className="mt-0"
          />
        </div>

        <div className="mt-2 px-4">
          <ul
            className="border border-gray-100 rounded-lg py-4"
            style={{
              boxShadow:
                "0px 4px 6px -2px #10182808, 0px 12px 16px -4px #10182814",
            }}
          >
            {topLinks.map((topLink, idx) => (
              <li key={idx}>
                <a
                  href="/"
                  className="flex justify-between items-center px-3 py-2 transition hover:bg-gray-50"
                >
                  <div className="flex">
                    <img
                      src={topLink.icon}
                      alt=""
                      className="inline-block mr-3"
                    />
                    <em className="not-italic font-medium text-gray-700">
                      {topLink.title}
                    </em>
                  </div>
                  {topLink.count && (
                    <div className="w-9 h-6 bg-gray-100 flex justify-center items-center rounded-2xl font-medium text-sm text-gray-700">
                      {topLink.count}
                    </div>
                  )}
                </a>
              </li>
            ))}
            <li className="font-medium text-gray-700 px-3 py-2">Drew Cano</li>
          </ul>
        </div>
        <div className="grow lg:hidden" />
        <div className="border-b border-gray-200 pb-6 mx-4 lg:mt-4 lg:border-0">
          <ul>
            {bottomLinks.map((bottomLink, idx) => (
              <li key={idx}>
                <a
                  href="/"
                  className="flex justify-between items-center px-3 py-2 transition hover:bg-gray-50"
                >
                  <div className="flex">
                    <img
                      src={bottomLink.icon}
                      alt=""
                      className="inline-block mr-3"
                    />
                    <em className="not-italic font-medium text-gray-700">
                      {bottomLink.title}
                    </em>
                  </div>
                </a>
              </li>
            ))}
          </ul>
        </div>
        <div className="bg-gray-50 rounded-lg mx-4 px-4 py-5  hidden lg:block">
          <h3 className="font-medium text-sm text-gray-900 mb-1">
            New features available!
          </h3>
          <em className="not-italic text-sm text-gray-500">
            Check out the new dashboard view. Pages now load faster
          </em>
          <img
            className="inline-block rounded-[6px] my-4"
            src={navImage}
            alt=""
          />
          <div>
            <button className="font-medium text-sm text-gray-500 mr-3 transition hover:text-gray-400">
              Dismiss
            </button>
            <button className="font-medium text-sm text-primary-700 transition hover:text-primary-900">
              What's new?
            </button>
          </div>
        </div>
        <div className="flex items-start mt-6 px-4 pb-8 lg:border-t lg:border-gray-200 lg:pt-6 lg:mx-4 lg:px-0">
          <img
            className="w-10 h-10 rounded-full inline-block mr-3"
            src={user1}
            alt=""
          />
          <div className="flex flex-col">
            <em className="not-italic font-medium text-gray-900 text-sm">
              Olivia Rhyne
            </em>
            <em className="not-italic text-sm text-gray-500">
              olivia@untitledui.com
            </em>
          </div>
          <div className="grow" />
          <button className="w-9 h-9 rounded-lg flex justify-center items-center transition hover:bg-gray-50">
            <img src={logout} alt="" />
          </button>
        </div>
      </aside>
    </>
  );
};
