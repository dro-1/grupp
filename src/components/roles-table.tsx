import arrowDown from "@/assets/icons/arrow-down.svg";
import primaryCheck from "@/assets/icons/primary-check.svg";
import greenCheck from "@/assets/icons/green-check.svg";
import lightDownload from "@/assets/icons/light-download.svg";

import { roles } from "@/utils/constants";
import { useState } from "react";
import { twMerge } from "tailwind-merge";
import { RoleUsersCard } from "./role-users-card";

export const RolesTable = () => {
  const [selectedRoles, setSelectedRoles] = useState<string[]>([]);
  return (
    <table
      className="w-full border border-gray-200 max-w-[700px] min-[700px]:ml-4 lg:ml-8 lg:max-w-[unset] lg:w-[calc(100%-64px)]"
      style={{
        boxShadow: "0px 2px 4px -2px #1018280F, 0px 4px 8px -2px #1018281A",
      }}
    >
      <thead>
        <tr className="bg-gray-50 border-y border-gray-200 py-3 px-6">
          <th className="py-3 px-6">
            <div className="flex items-center group">
              <div
                className={twMerge(
                  "w-5 h-5 bg-white cursor-pointer border border-gray-300 rounded-[6px] mr-3 flex justify-center items-center transition hover:border-primary-600 hover:bg-primary-50",
                  selectedRoles.length == roles.length &&
                    "border-primary-600 bg-primary-50"
                )}
                onClick={() => {
                  if (selectedRoles.length != roles.length) {
                    setSelectedRoles(roles.map((role) => role.name));
                  } else {
                    setSelectedRoles([]);
                  }
                }}
              >
                <img
                  src={primaryCheck}
                  alt=""
                  className={twMerge(
                    "w-[12px] hidden",
                    selectedRoles.length == roles.length && "block"
                  )}
                />
              </div>
              <span className="font-medium text-xs tracking-0 text-gray-500 transition group-hover:text-gray-700">
                Name
              </span>
              <img src={arrowDown} alt="" className="inline-block ml-1" />
            </div>
          </th>
          <th className="py-3 px-6 font-medium text-xs tracking-0 text-gray-500 text-left sm:table-cell transition hover:text-gray-700">
            Type
          </th>
          <th className="py-3 px-6 font-medium text-xs tracking-0 text-gray-500 text-left transition hover:text-gray-700">
            Date Created
          </th>
          <th className="py-3 px-6 font-medium text-xs tracking-0 text-gray-500 text-left hidden md:table-cell transition hover:text-gray-700">
            Status
          </th>
          <th className="py-3 px-6 font-medium text-xs tracking-0 text-gray-500 text-left hidden xl:table-cell transition hover:text-gray-700">
            Role Users
          </th>
          <th className="py-3 px-6 font-medium text-xs tracking-0 text-gray-500 text-left hidden xl:table-cell"></th>
        </tr>
      </thead>
      <tbody>
        {roles.map((role, idx) => (
          <tr
            key={idx}
            className="bg-white border-y border-gray-200 py-3 px-6 transition hover:bg-gray-50"
          >
            <td className="py-6 px-6">
              <div className="flex items-center">
                <div
                  className={twMerge(
                    "w-5 h-5 bg-white border cursor-pointer border-gray-300 rounded-[6px] mr-3 flex justify-center items-center transition hover:border-primary-600 hover:bg-primary-50",
                    selectedRoles.includes(role.name) &&
                      "border-primary-600 bg-primary-50"
                  )}
                  onClick={() => {
                    if (selectedRoles.includes(role.name)) {
                      setSelectedRoles((selectedRoles) =>
                        selectedRoles.filter((iRole) => iRole != role.name)
                      );
                    } else {
                      setSelectedRoles((selectedRoles) => [
                        ...selectedRoles,
                        role.name,
                      ]);
                    }
                  }}
                >
                  <img
                    src={primaryCheck}
                    alt=""
                    className={twMerge(
                      "w-[12px] hidden",
                      selectedRoles.includes(role.name) && "block"
                    )}
                  />
                </div>
                <span className="font-medium text-sm tracking-0 text-gray-900">
                  {role.name}
                </span>
              </div>
            </td>
            <td className="py-6 px-6 text-sm tracking-0 text-gray-500 text-left hidden sm:table-cell">
              {role.type}
            </td>
            <td className="py-6 px-6 text-sm tracking-0 text-gray-500 text-left">
              {role.date}
            </td>
            <td className="py-6 px-6 text-sm tracking-0 text-gray-500 text-left hidden md:table-cell">
              {role.status == "ACTIVE" && (
                <div className="flex items-center justify-center w-[67px] h-[22px] rounded-2xl font-medium text-xs tracking-0 bg-success-50 text-success-700">
                  <img src={greenCheck} alt="" className="inline-block mr-1" />
                  Active
                </div>
              )}
              {role.status == "INACTIVE" && (
                <div className="flex items-center justify-center w-[67px] h-[22px] rounded-2xl font-medium text-xs tracking-0 bg-orange text-gray-6">
                  Inactive
                </div>
              )}
            </td>
            <td className="py-6 px-6 text-sm tracking-0 text-gray-500 text-left hidden xl:table-cell">
              <RoleUsersCard images={role.users} />
            </td>
            <td className="py-6 px-6 hidden xl:table-cell">
              <button>
                <img src={lightDownload} alt="" />
              </button>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};
