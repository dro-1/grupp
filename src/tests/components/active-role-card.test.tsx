import { ActiveRoleCard } from "@/components/active-role-card";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Active Role Card", () => {
  it("renders correctly", () => {
    const tree = render(
      <ActiveRoleCard
        activeRole={{
          id: "001",
          lastActiveDate: "06/2023",
          roleName: "Superadmin",
        }}
        selected={false}
        onClick={() => {}}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("changes background color when selected", async () => {
    const user = userEvent.setup();
    const handleClick = () => {
      rerender(
        <ActiveRoleCard
          activeRole={{
            id: "001",
            lastActiveDate: "06/2023",
            roleName: "Superadmin",
          }}
          selected={true}
          onClick={handleClick}
        />
      );
    };
    const { getByTestId, rerender } = render(
      <ActiveRoleCard
        activeRole={{
          id: "001",
          lastActiveDate: "06/2023",
          roleName: "Superadmin",
        }}
        selected={false}
        onClick={handleClick}
      />
    );
    const card = getByTestId("active-role-card-001");
    expect(card.className).not.toContain("border-primary-300 bg-primary-50");
    await user.click(card);
    expect(card.className).toContain("border-primary-300 bg-primary-50");
  });

  it("changes background color when unselected", async () => {
    const user = userEvent.setup();
    const handleClick = () => {
      rerender(
        <ActiveRoleCard
          activeRole={{
            id: "001",
            lastActiveDate: "06/2023",
            roleName: "Superadmin",
          }}
          selected={false}
          onClick={handleClick}
        />
      );
    };
    const { getByTestId, rerender } = render(
      <ActiveRoleCard
        activeRole={{
          id: "001",
          lastActiveDate: "06/2023",
          roleName: "Superadmin",
        }}
        selected={true}
        onClick={handleClick}
      />
    );
    const card = getByTestId("active-role-card-001");
    expect(card.className).toContain("border-primary-300 bg-primary-50");
    await user.click(card);
    expect(card.className).not.toContain("border-primary-300 bg-primary-50");
  });

  it("changes text color when selected", async () => {
    const user = userEvent.setup();
    const activeRole = {
      id: "001",
      lastActiveDate: "06/2023",
      roleName: "Superadmin",
    };
    const handleClick = () => {
      rerender(
        <ActiveRoleCard
          activeRole={activeRole}
          selected={true}
          onClick={handleClick}
        />
      );
    };
    const { getByTestId, rerender, getByText } = render(
      <ActiveRoleCard
        activeRole={activeRole}
        selected={false}
        onClick={handleClick}
      />
    );
    const card = getByTestId("active-role-card-001");
    const roleName = getByText("Superadmin");
    const lastActiveDate = getByText(
      "Last active " + activeRole.lastActiveDate
    );
    const setDefault = getByText("Set as default");
    expect(roleName).not.toHaveClass("text-primary-800");
    expect(lastActiveDate).not.toHaveClass("text-primary-600");
    expect(setDefault).not.toHaveClass("text-primary-500");
    await user.click(card);
    expect(roleName).toHaveClass("text-primary-800");
    expect(lastActiveDate).toHaveClass("text-primary-600");
    expect(setDefault).toHaveClass("text-primary-500");
  });
  it("changes checkmark background when selected", async () => {
    const user = userEvent.setup();
    const activeRole = {
      id: "001",
      lastActiveDate: "06/2023",
      roleName: "Superadmin",
    };
    const handleClick = () => {
      rerender(
        <ActiveRoleCard
          activeRole={activeRole}
          selected={true}
          onClick={handleClick}
        />
      );
    };
    const { getByTestId, rerender } = render(
      <ActiveRoleCard
        activeRole={activeRole}
        selected={false}
        onClick={handleClick}
      />
    );
    const card = getByTestId("active-role-card-001");
    const checkmarkContainer = getByTestId("checkmark-container");
    expect(checkmarkContainer.className).not.toContain(
      "bg-primary-600 border-primary-600"
    );
    await user.click(card);
    expect(checkmarkContainer.className).toContain(
      "bg-primary-600 border-primary-600"
    );
  });
});
