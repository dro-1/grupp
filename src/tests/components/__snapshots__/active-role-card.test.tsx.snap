// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Active Role Card > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex p-4 bg-white border cursor-pointer border-gray-200 rounded-lg max-w-[784px]"
        data-testid="active-role-card-001"
      >
        <div
          class="bg-white w-[45px] h-[31px] border border-[#f2f4f7] flex justify-center items-center rounded-[6px] mr-3"
        >
          <img
            alt=""
            src="data:image/svg+xml,%3csvg%20width='24'%20height='20'%20viewBox='0%200%2024%2020'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M17%2019V17C17%2015.9391%2016.5786%2014.9217%2015.8284%2014.1716C15.0783%2013.4214%2014.0609%2013%2013%2013H5C3.9391%2013%202.9217%2013.4214%202.1716%2014.1716C1.4214%2014.9217%201%2015.9391%201%2017V19M23%2019V17C22.9993%2016.1137%2022.7044%2015.2528%2022.1614%2014.5523C21.6184%2013.8519%2020.8581%2013.3516%2020%2013.13M16%201.13C16.8604%201.3503%2017.623%201.8507%2018.1676%202.55231C18.7122%203.25392%2019.0078%204.1168%2019.0078%205.005C19.0078%205.8932%2018.7122%206.7561%2018.1676%207.4577C17.623%208.1593%2016.8604%208.6597%2016%208.88M13%205C13%207.2091%2011.2091%209%209%209C6.7909%209%205%207.2091%205%205C5%202.79086%206.7909%201%209%201C11.2091%201%2013%202.79086%2013%205Z'%20stroke='%23667085'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
          />
        </div>
        <div
          class="grow tracking-0 flex flex-col"
        >
          <em
            class="not-italic font-medium text-sm text-gray-700"
          >
            Superadmin
          </em>
          <em
            class="not-italic text-[#70798C] text-sm"
          >
            Last active 
            06/2023
          </em>
          <div
            class="mt-2"
          >
            <button
              class="text-gray-500 font-medium text-sm mr-3 transition lg:hover:text-gray-700"
            >
              Set as default
            </button>
            <button
              class="text-primary-700 font-medium text-sm transition lg:hover:text-primary-900"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="w-4 h-4 bg-white border border-gray-300 rounded-full flex justify-center items-center transition"
          data-testid="checkmark-container"
        >
          <img
            alt=""
            src="data:image/svg+xml,%3csvg%20width='10'%20height='10'%20viewBox='0%200%2010%2010'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M8.33329%202.5L3.74996%207.08333L1.66663%205'%20stroke='white'%20stroke-width='1.66667'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex p-4 bg-white border cursor-pointer border-gray-200 rounded-lg max-w-[784px]"
      data-testid="active-role-card-001"
    >
      <div
        class="bg-white w-[45px] h-[31px] border border-[#f2f4f7] flex justify-center items-center rounded-[6px] mr-3"
      >
        <img
          alt=""
          src="data:image/svg+xml,%3csvg%20width='24'%20height='20'%20viewBox='0%200%2024%2020'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M17%2019V17C17%2015.9391%2016.5786%2014.9217%2015.8284%2014.1716C15.0783%2013.4214%2014.0609%2013%2013%2013H5C3.9391%2013%202.9217%2013.4214%202.1716%2014.1716C1.4214%2014.9217%201%2015.9391%201%2017V19M23%2019V17C22.9993%2016.1137%2022.7044%2015.2528%2022.1614%2014.5523C21.6184%2013.8519%2020.8581%2013.3516%2020%2013.13M16%201.13C16.8604%201.3503%2017.623%201.8507%2018.1676%202.55231C18.7122%203.25392%2019.0078%204.1168%2019.0078%205.005C19.0078%205.8932%2018.7122%206.7561%2018.1676%207.4577C17.623%208.1593%2016.8604%208.6597%2016%208.88M13%205C13%207.2091%2011.2091%209%209%209C6.7909%209%205%207.2091%205%205C5%202.79086%206.7909%201%209%201C11.2091%201%2013%202.79086%2013%205Z'%20stroke='%23667085'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
        />
      </div>
      <div
        class="grow tracking-0 flex flex-col"
      >
        <em
          class="not-italic font-medium text-sm text-gray-700"
        >
          Superadmin
        </em>
        <em
          class="not-italic text-[#70798C] text-sm"
        >
          Last active 
          06/2023
        </em>
        <div
          class="mt-2"
        >
          <button
            class="text-gray-500 font-medium text-sm mr-3 transition lg:hover:text-gray-700"
          >
            Set as default
          </button>
          <button
            class="text-primary-700 font-medium text-sm transition lg:hover:text-primary-900"
          >
            Edit
          </button>
        </div>
      </div>
      <div
        class="w-4 h-4 bg-white border border-gray-300 rounded-full flex justify-center items-center transition"
        data-testid="checkmark-container"
      >
        <img
          alt=""
          src="data:image/svg+xml,%3csvg%20width='10'%20height='10'%20viewBox='0%200%2010%2010'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M8.33329%202.5L3.74996%207.08333L1.66663%205'%20stroke='white'%20stroke-width='1.66667'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
